src/serializer/ResidualHeapSerializer.js

```diff
@@ -1197,7 +1208,6 @@ export class ResidualHeapSerializer {
       },
     };
 
-    invariant(this.serializedValues.size === this.residualValues.size);
```

Comment:
I haven't figured out an efficient way to calculate this number in the general case because in the case of `Date.now` if we remove n derivedIds, we remove 2n elements from serializedValues (because `Date.now` has a dummy `StringValue("__Date.now()")` argument). In general removing a generator entry removes the need to serialize its arguments which may or may not be needed by something else.